yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -548,8 +596,13 @@ private[spark] class Client(
     logInfo("Setting up container launch context for our AM")
     val appId = newAppResponse.getApplicationId
     val appStagingDir = getAppStagingDir(appId)
-    val localResources = prepareLocalResources(appStagingDir)
     val launchEnv = setupLaunchEnv(appStagingDir)
+    val localResources = prepareLocalResources(appStagingDir)
+
+    // Set the environment variables to be passed on to the executors.
+    distCacheMgr.setDistFilesEnv(launchEnv)
+    distCacheMgr.setDistArchivesEnv(launchEnv)
```

Comment:
I noticed that we order some of these code blocks around in this patch. Is there a particular reason for it? Have you verified that it's safe to do so?